### PR TITLE
addressed the issues with client losing connectivity

### PR DESCRIPTION
**Pull request by iicaisin1**
addressed the issues with client losing connectivity after a longer period of time

### CODE REVIEW

**Comment by vik-borisov** [comment]
@iicaisin1 
Hello, please remove your code style from pool request

**Comment by iicaisin1** [comment]
@vik-borisov , 
Hi,  I removed the indentation with tabs. I also added the editorconfig for others to automatically follow the pattern.

File: src/TelegramClient.Core/Network/Tcp/TcpService.cs [hunk 1]
```diff
@@ -69,19 +73,16 @@ private void Dispose(bool disposing)
                     await _semaphore.WaitAsync();
                     var endpoint = (IPEndPoint)_tcpClient.Client.RemoteEndPoint;
 
-                    if (!_tcpClient.Connected || endpoint.Address.ToString() != session.ServerAddress || endpoint.Port != session.Port)
+                    if (!_tcpClient.Connected || _tcpClient.Client == null || !_tcpClient.Client.Connected || endpoint.Address.ToString() != session.ServerAddress || endpoint.Port != session.Port)
                     {
-                        if (!_tcpClient.Connected || endpoint.Address.ToString() != session.ServerAddress || endpoint.Port != session.Port)
+                        if (_tcpClient != null)
```
[the remark this answers is not in the document]
Comment (vik-borisov) [reply src/TelegramClient.Core/Network/Tcp/TcpService.cs]: @iicaisin1 please remove duplicate condition

File: src/TelegramClient.Core/Network/Tcp/TcpTransport.cs [hunk 2]
```diff
@@ -117,19 +117,23 @@ private void PushToQueue(byte[] packet, TaskCompletionSource<bool> tcs)
         {
             _queue.Enqueue(Tuple.Create(packet, tcs));
 
-            _semaphoreSlim.WaitAsync().ContinueWith(_ =>
+            Task.Run(async () =>
             {
-                if (!_queue.IsEmpty)
```
Comment (vik-borisov) [inline src/TelegramClient.Core/Network/Tcp/TcpTransport.cs]: may be extract method and don't use Task.Run

**Pull request closed without merging**